src/api/message_builder/input_selection/native_token_helpers.rs

```diff
@@ -56,7 +56,7 @@ pub(crate) fn get_minted_and_melted_native_tokens(
             let TokenScheme::Simple(output_foundry_simple_ts) = output_foundry.token_scheme();
             let mut initial_creation = true;
 
-            for input in inputs {
+            for input in inputs.clone() {
```

Comment:
Do we really need to clone here?

Reply:
Yes but it's just cloning a pointer

Reply:
yes because this is inside a loop that starts at line `54` but silly github doesn't show it.